tests/sentry/replays/test_organization_replay_index.py

```diff
@@ -938,9 +947,27 @@ def test_get_replays_filter_clicks(self):
                 "replay_click.testid:2",
                 "replay_click.textContent:World",
                 "replay_click.title:NotMyTitle",
+                "!replay_click.selector:div#myid",
+                "replay_click.selector:div#notmyid",
+                # Assert all classes must match.
+                "replay_click.selector:div#myid.class1.class2.class3",
+                # Invalid selectors return no rows.
+                "replay_click.selector:$#%^#%",
+                # Integer type role values are not allowed and must be wrapped in single quotes.
+                "replay_click.selector:div[title=1]",
             ]
             for query in queries:
                 response = self.client.get(self.url + f"?query={query}")
                 assert response.status_code == 200, query
                 response_data = response.json()
                 assert len(response_data["data"]) == 0, query
+
+    def test_get_replays_filter_clicks_nested_selector(self):
+        """Test replays do not support nested selectors."""
+        project = self.create_project(teams=[self.team])
+        self.store_replays(mock_replay(datetime.datetime.now(), project.id, uuid.uuid4().hex))
+
+        with self.feature(REPLAYS_FEATURES):
+            response = self.client.get(self.url + '?field=id&query=replay_click.selector:".a .b"')
```

Comment:
could we add tests for further unsupported selector syntax? or do let me know if any of these are supported:

- `:not(p)`
- `a:visited`
- `::selection`
- `a[href^="https"]`
- `p ~ ul`

going off list here https://www.w3schools.com/cssref/css_selectors.php

Reply:
@JoshFerge Locked down the rest of the syntax and added coverage for marshaling the errors.